drivers/spi/spim_nrf52.c

```diff
@@ -324,13 +324,13 @@ static int spim_nrf52_init(struct device *dev)
 
 	for (i = 0; i < sizeof(config->psel.ss); i++) {
 		if (config->psel.ss[i] != SS_UNUSED) {
+			spim_nrf52_csn(data->gpio_port, config->psel.ss[i],
```

Comment:
Please revert to setting GPIO config prior to SPIM selection, but use a correct pullup/pulldown setting.

Refer to how it is for UART... https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/serial/uart_nrf5.c#L205

The reason is, all peripherals in nRF5 when configured will internally grab the GPIO and place the GPIO line in required state by that protocol. This is the reason why moving gpio_pin_configure after spim csn call solves your problem, as gpio configure has no effect after spim has grabbed the line. When the peripheral is powered down or SoC enters system off, GPIO configuration is reverted back to (requiring correct pullup/pulldown to be set).